app/tasks/raster_tile_cache_assets.py

```diff
@@ -0,0 +1,423 @@
+    # zoom levels, so start at the highest and work our way back
+
+    co = copy.deepcopy(source_creation_options)
+    co["resampling"] = "med"
+    co["overwrite"] = co.get("overwrite", False)
+    co["srid"] = "epsg-3857"
+
+    for zoom_level in reversed(range(min_zoom, max_zoom)):
+        co["grid"] = f"zoom_{zoom_level}"
+
+        asset_uri = get_asset_uri(dataset, version, AssetType.raster_tile_set, co)
+        del co["srid"]
+
+        co_obj = RasterTileSetSourceCreationOptions(**co)
+
+        # Create an asset record
+        asset_options = AssetCreateIn(
+            asset_type=AssetType.raster_tile_set,
+            asset_uri=asset_uri,
+            is_managed=True,
+            creation_options=co_obj,
+            metadata={},
+        ).dict(by_alias=True)
+        wm_asset_record = await create_asset(dataset, version, **asset_options)
+        print(f"ZOOM LEVEL {zoom_level} REPROJECTION ASSET CREATED")
+
+        zoom_level_job = await _run_pixetl(
+            dataset,
+            version,
+            wm_asset_record.creation_options,
+            f"zoom_level_{zoom_level}_{co['pixel_meaning']}_reprojection",
+            callback_constructor(wm_asset_record.asset_id),
+            parents=(parents + reprojection_jobs) if parents else reprojection_jobs,
+        )
+        reprojection_jobs.append(zoom_level_job)
+        print(f"ZOOM LEVEL {zoom_level} REPROJECTION JOB CREATED")
+
+        co["srid"] = "epsg-3857"
+        source_uri = get_asset_uri(
+            dataset, version, AssetType.raster_tile_set, co
+        ).replace("{tile_id}.tif", "tiles.geojson")
+        co["source_uri"] = [source_uri]
+
+    return reprojection_jobs
+
+
+async def _merge_intensity_and_date_conf(
+    dataset: str,
+    version: str,
+    date_conf_co: Dict[str, Any],
+    intensity_co: Dict[str, Any],
+    min_zoom: int,
+    max_zoom: int,
+    parents: List[Job],
+):
+    merge_intensity_jobs: List[Job] = []
+
+    for zoom_level in range(min_zoom, max_zoom):
+        # Sanitize creation_options
+
+        d_c_co = copy.deepcopy(date_conf_co)
+        d_c_co["srid"] = "epsg-3857"
+        d_c_co["grid"] = f"zoom_{zoom_level}"
+        date_conf_uri = get_asset_uri(
+            dataset, version, AssetType.raster_tile_set, d_c_co
+        ).replace("{tile_id}.tif", "tiles.geojson")
+
+        i_co = copy.deepcopy(intensity_co)
+        i_co["srid"] = "epsg-3857"
+        i_co["grid"] = f"zoom_{zoom_level}"
+        intensity_uri = get_asset_uri(
+            dataset, version, AssetType.raster_tile_set, i_co
+        ).replace("{tile_id}.tif", "tiles.geojson")
+
+        c_co = copy.deepcopy(intensity_co)
+        c_co["srid"] = "epsg-3857"
+        c_co["grid"] = f"zoom_{zoom_level}"
+        c_co["pixel_meaning"] = RGB_ENCODED_PIXEL_MEANING
+        asset_uri = get_asset_uri(dataset, version, AssetType.raster_tile_set, c_co)
+        merged_asset_prefix = asset_uri.rsplit("/", 1)[0]
```

Comment:
This is the same as os.path.dirname right? I believe the this would be easier to read, don't you think?